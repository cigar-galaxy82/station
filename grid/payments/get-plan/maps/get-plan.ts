/* eslint-disable jest/no-export */

import { describe, expect } from '@jest/globals';
import { SuperfaceTest } from '@superfaceai/testing';

import { createPlan } from '../../create-plan/maps/create-plan';

export function getPlanTest(providerName: string): void {
  describe(`payments/get-plan/${providerName}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest({
        profile: 'payments/get-plan',
        provider: providerName,
        testInstance: expect,
      });
    });

    describe('GetPlan', () => {
      describe('when plan exists', () => {
        it('gets a plan', async () => {
          const id = await createPlan(providerName);
          const result = await superface.run({
            useCase: 'GetPlan',
            input: {
              id,
            },
          });
          expect(() => result.unwrap()).not.toThrow();
          expect(result.unwrap()).toMatchSnapshot();
        });
      });
    });
  });
}
