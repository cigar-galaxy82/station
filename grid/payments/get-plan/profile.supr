name = "payments/get-plan"
version = "1.0.0"

"""
Get subscription plan
"""

usecase GetPlan safe {
  input {
    """
    ID
    Unique identifier of subscription plan.
    """
    id! string!
  }

  result {
    """
    Product ID
    Unique identifier of the product the plan is for (if applicable).
    """
    productId string
    
    """
    Name
    Name of the subscription plan.
    """
    name string!

    """
    Interval
    Frequency of the interval at which the customer is charged.

    Possible values:
    - `day`
    - `week`
    - `month`
    - `year`
    """
    interval enum {
      day, week, month, year
    }

    """
    Price
    Amount charged each `interval` cycle.
    """
    price number
    
    """
    Currency code
    Three-letter currency code in which `price` is specified, in ISO 4217 format.
    """
    currency string

    """
    State
    State of the plan.

    Possible values:
     - `created` - The plan was just created. It may not be possible to accept new subscriptions yet.
     - `active` - The plan is active and accepting new subscriptions.
     - `inactive` - The plan has been deactivated, and cannot accept new subscriptions.
    """
    state enum {
      created, active, inactive
    }
  }


  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title! string

    """
    Detail
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail string
  }

  example Successful {
    input {
      id = "PLAN-89101829211"
    }

    result {
      productId = "PROD-1192810198"
      name = "Monthly subscription to Superface milk delivery"
      interval = "month"
      price = 25
      currency = "USD"
      state = "active"
    }
  }
}