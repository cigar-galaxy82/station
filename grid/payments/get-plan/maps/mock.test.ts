import { getPlanTest } from './get-plan';

getPlanTest('mock');
