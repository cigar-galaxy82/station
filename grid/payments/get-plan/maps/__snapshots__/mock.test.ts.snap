// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments/get-plan/mock GetPlan when plan exists gets a plan 1`] = `
Object {
  "currency": "USD",
  "interval": "month",
  "name": "Monthly subscription to Superface milk delivery",
  "price": 25,
  "productId": "PROD-1192810198",
  "state": "active",
}
`;
